src/core/ext/xds/xds_cluster.cc

```diff
@@ -398,6 +400,108 @@ grpc_error_handle CdsResourceParse(
       }
     }
   }
+  // As long as outlier detection field is present in the cluster update,
+  // we will end up with a outlier detection in the cluster resource which will
+  // lead to the creation of outlier detection in discovery mechanism.  Values
+  // for outlier detection will be based on fields received and
+  // default values.
+  if (envoy_config_cluster_v3_Cluster_has_outlier_detection(cluster)) {
```

Comment:
I just noticed that we're missing the env var guard here.  We'll need to add that before we can merge this.